frontend/src/components/v2/SecretInput/SecretInput.tsx

```diff
@@ -1,10 +1,15 @@
 /* eslint-disable react/no-danger */
-import { forwardRef, TextareaHTMLAttributes } from "react";
+import React, { forwardRef, TextareaHTMLAttributes, useRef, useState } from "react";
 import { twMerge } from "tailwind-merge";
 
+import {
+  REGEX_SECRET_REFERENCE_FIND,
```

Comment:
Avoid distributing the logics into so many files. This logic is never shared anywhere other than mostly thing component so keep the regex here, co-location functions is easier to manage

Reply:
Hey @akhilmhdh!  this regex is shared among the helper function `isValidSecretReferenceValue` and the `InfisicalSecretInput` component. 

The `isValidSecretReferenceValue` is used inside the secrets mutation file (hook). Would you happen to have a recommendation for how to structure this instead? 

![image](https://github.com/Infisical/infisical/assets/65645666/dd523e09-18d9-4e8c-a041-4c89218016c9)
![image](https://github.com/Infisical/infisical/assets/65645666/8829a7be-21b7-4c87-9537-3eb773928563)